File: lib/src/main/java/com/sd/lib/webview/FWebViewHandler.java
package com.sd.lib.webview;

import android.webkit.WebView;

import java.net.HttpCookie;
import java.util.List;

public abstract class FWebViewHandler {
    /**
     * 初始化webview
     */
    public abstract void onInitWebView(WebView webView);

    /**
     * 返回http的cookie
     */
    public abstract List<HttpCookie> getHttpCookie(String url);

    /**
     * 把webview的cookie同步到http
     */
    public abstract void synchronizeWebViewCookieToHttp(String url, List<HttpCookie> listCookie);
}

File: lib/src/main/java/com/sd/lib/webview/FWebViewHandler.kt
package com.sd.lib.webview

import android.webkit.WebView
import java.net.HttpCookie

abstract class FWebViewHandler {
    /**
     * 初始化webview
     */
    abstract fun initWebView(webView: WebView)

    /**
     * 返回http的cookie
     */
    abstract fun getHttpCookie(url: String): List<HttpCookie>?

    /**
     * 把webview的cookie同步到http
     */
    abstract fun synchronizeWebViewCookieToHttp(url: String, listCookie: List<HttpCookie>)
}